cinder/objects/snapshot.py

```diff
@@ -29,6 +29,18 @@
 
 CONF = cfg.CONF
 
+sap_custom_opts = [
+    cfg.BoolOpt('sap_allow_independent_snapshots',
+                default=False,
+                help='Allow cinder to schedule snapshot creations on pools '
+                     'other than the source volume pool.'),
+    cfg.BoolOpt('sap_allow_independent_clone',
+                default=False,
+                help='Allow cinder to schedule a clone volume on a pool '
+                     'other than the source volume pool.'),
+]
+CONF.register_opts(sap_custom_opts)
```

Comment:
No other object registers config options. Are you sure this is the right place to do so?

You could create a `cinder.common.sap` where you put the `sap_custom_opts` and then you register them in the scheduler's manager and in the volume manager (instead of relying on auto-register-through-import of an otherwise unused module).

While you're at it, you could add its own ini section `[SAP]` and access things `CONF.SAP.allow_independent_snapshots`. :D

Reply:
I was just looking for an existing common area that is included already by both the scheduler and the volume manager to ensure that the config options are always registered since they are need by both the driver and the scheduler.  I think it's a bit hair splitting where the common config options exist.  The snapshot.py made sense since the config option were related to a snapshot.

The advantage of creating a common.sap.py is that it will be easier to rebase, since it's a new file that doesn't exist upstream.  I like your suggestion and I'll create a common/sap.py for this.